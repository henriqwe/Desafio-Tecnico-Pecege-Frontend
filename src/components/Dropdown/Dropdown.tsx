import * as DropdownMenu from '@radix-ui/react-dropdown-menu'
import './style.css'
import { EllipsisVerticalIcon } from '@heroicons/react/24/outline'
import { ReactNode } from 'react'

interface props {
  options: { fn?: () => void; content: ReactNode }[]
}
export function Dropdown({ options }: props) {
  return (
    <DropdownMenu.Root>
      <DropdownMenu.Trigger asChild data-testid={'dropdown-menu-trigger'}>
        <button className="IconButton">
          <EllipsisVerticalIcon />
        </button>
      </DropdownMenu.Trigger>

      <DropdownMenu.Portal>
        <DropdownMenu.Content className="DropdownMenuContent" sideOffset={5}>
          {options.map((option, idx) => (
            <DropdownMenu.Item
              className="DropdownMenuItem"
              onClick={() => option.fn?.()}
              key={'DropdownMenuItem' + idx}
            >
              {option.content}
            </DropdownMenu.Item>
          ))}
        </DropdownMenu.Content>
      </DropdownMenu.Portal>
    </DropdownMenu.Root>
  )
}
