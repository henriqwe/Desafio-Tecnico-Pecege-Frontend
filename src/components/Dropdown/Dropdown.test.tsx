import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Dropdown } from './Dropdown'

const FnDropdownOption = jest.fn()
const dropdownOption = [
  {
    content: <div data-testeid={'dropdownoption-content'}>test</div>,
    fn: () => FnDropdownOption()
  }
]

describe('Dropdown', () => {
  it('should render correctly snapshot', async () => {
    const { container } = render(<Dropdown options={dropdownOption} />)

    expect(container).toMatchSnapshot()
  })
})
