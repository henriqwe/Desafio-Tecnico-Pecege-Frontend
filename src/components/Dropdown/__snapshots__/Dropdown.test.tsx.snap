// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render correctly snapshot 1`] = `
<div>
  <button
    aria-expanded="false"
    aria-haspopup="menu"
    class="IconButton"
    data-state="closed"
    data-testid="dropdown-menu-trigger"
    id="radix-:r0:"
    type="button"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 6.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 12.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 18.75a.75.75 0 110-1.5.75.75 0 010 1.5z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</div>
`;
